core/schema/sdk.go

```diff
@@ -633,6 +633,34 @@ func (sdk *goSDK) baseWithCodegen(
 				},
 			},
 		},
+	}
+
+	// inject sdk specific env variables before withExec
+	// for now allow only Env with prefix GO
+	cfg, ok, _ := src.Self.ModuleConfig(ctx)
+	if ok && cfg.SDK != nil {
+		for k, v := range cfg.SDK.Env {
+			if !strings.HasPrefix(k, "GO") {
+				continue
+			}
+
+			selectors = append(selectors, dagql.Selector{
+				Field: "withEnvVariable",
+				Args: []dagql.NamedInput{
+					{
+						Name:  "name",
+						Value: dagql.NewString(k),
+					},
+					{
+						Name:  "value",
```

Comment:
@jedevc, @marcosnils  I have been thinking a bit more about this specific code here. I think the runtime here will use the environment variable specified in the `top-level` module's SDK, but what do you think should happen if a dependency exposes an environment variable in SDK config?

Reply:
Bit confused by this - if a dependency exposes an env variable in it's own config, then *it's* runtime should get that environment variable. But the top-level module shouldn't be affected at all.

Reply:
ah. I keep forgetting deps get their own runtime/server etc. thanks.